lib/fluent/plugin/in_s3.rb

```diff
@@ -90,6 +90,8 @@ def initialize
     config_param :check_apikey_on_start, :bool, default: true
     desc "URI of proxy environment"
     config_param :proxy_uri, :string, default: nil
+    desc "Optional RegEx to match incoming messages"
+    config_param :match_regexp, :string, default: nil
```

Comment:
Could you use `:regexp` type like this?
https://github.com/fluent/fluentd/blob/f81d60f84cc5fd569c1675f43dcd240cee0a05b6/lib/fluent/plugin/parser.rb#L92

It will check and compile the expression automatically.